spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceInitializer.java

```diff
@@ -125,10 +125,7 @@ private boolean isEnabled() {
 		if (mode == DataSourceInitializationMode.NEVER) {
 			return false;
 		}
-		if (mode == DataSourceInitializationMode.EMBEDDED && !isEmbedded()) {
-			return false;
-		}
-		return true;
+		return mode != DataSourceInitializationMode.EMBEDDED || isEmbedded();
```

Comment:
When there are multiple if statements like this, I find the current version more readable than collapsing the last if statement as you suggest.